internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -230,6 +230,18 @@ func (r *Reconciler) reconcileControlPlane(ctx context.Context, s *scope.Scope)
 		return err
 	}
 
+	// If the controlPlane has infrastructureMachines and the InfrastructureMachineTemplate has changed on this reconcile
+	// delete the old template.
+	// This is a best effort deletion only and may leak templates if an error occurs during reconciliation.
+	// TODO:(killianmuldoon) Make a more holistic template deletion flow.
```

Comment:
I'll create an issue covering over the problems with the way we currently handle template rotation.

Reply:
Maybe we shouldn't have this todo here (when we merge this PR). I think it's better to just have an issue describing the holistic problem vs having the TODO here specifically (as the problem is much more widespread then just this location)